antarest/study/storage/variantstudy/model/command/update_link.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+import typing as t
+
+from pydantic import ValidationError
+
+from antarest.core.exceptions import LinkValidationError
+from antarest.study.storage.rawstudy.model.filesystem.config.model import FileStudyTreeConfig
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.variantstudy.model.command.common import CommandName, CommandOutput
+from antarest.study.storage.variantstudy.model.command.create_link import AbstractLinkCommand, LinkProperties
+from antarest.study.storage.variantstudy.model.command.icommand import MATCH_SIGNATURE_SEPARATOR, ICommand, OutputTuple
+from antarest.study.storage.variantstudy.model.model import CommandDTO
+
+
+class UpdateLink(AbstractLinkCommand):
+    """
+    Command used to update a link between two areas.
+    """
+
+    # Overloaded metadata
+    # ===================
+
+    command_name: CommandName = CommandName.UPDATE_LINK
+    version: int = 1
+
+    def _apply_config(self, study_data: FileStudyTreeConfig) -> OutputTuple:
+        area_from, area_to = sorted([self.area1, self.area2])
+
+        return (
+            CommandOutput(
+                status=True,
+                message=f"Link between '{self.area1}' and '{self.area2}' updated",
+            ),
+            {"area_from": area_from, "area_to": area_to},
+        )
+
+    def _apply(self, study_data: FileStudy, *a: str) -> CommandOutput:
+        version = study_data.config.version
+        area_from, area_to = sorted([self.area1, self.area2])
+
+        try:
+            properties = LinkProperties.model_validate(self.parameters or {}).model_dump(
+                exclude_unset=True, by_alias=True
+            )
+        except ValidationError:
+            raise LinkValidationError("One or more fields are forbidden")
```

Comment:
I'm not sure we should catch this error. Because model_validate will already raise the `ValidationError` which has all info we need and the correct HTTP code I believe. At least you should do this to keep the stacktrace but I prefer not catching:
```python
except ValidationError as e:
     raise LinkValidationError("One or more fields are forbidden") from e
```

Reply:
I removed the catch